### PR TITLE
Ensure all parameters are documented

### PR DESCRIPTION
**Pull request by smortex**
The module has a lot of puppet strings but some parameters lack documentation.  This PR intends to add the missing documentation along with tooling to alert contributors when such documentation is missing.

Current status:
* [X] Match actual parameters order for parameters documentation
* [X] Align documentation vertically for better legibility
* [X] Add a unit test that ensures puppet-strings is happy
* [X] Remove dashed between parameter name and documentation (output to the documentation otherwise)
* [x] Add the missing documentation

### CODE REVIEW

**Comment by jflorian** [comment]
Just dropping in real quick to offer an immense THANK YOU for this.  Not that things were that bad, but this is pure goodness that's usually not the most fun thing to do.

File: manifests/storage.pp [hunk 1]
```diff
@@ -1,23 +1,25 @@
...
-# @param rundir
-# @param services
-# @param storage
-# @param user
+# @param maxconcurjobs  maximum number of Jobs that may run concurrently
+# @param media_type     The type of media supported by this device
+# @param password       Specifies the password that must be supplied by the named Director
+# @param port           The listening port for the Storage Daemon
+# @param rundir         The directory in which the Director may put its process Id file files
+# @param storage        FIXME: Unused?
```
Comment (smortex) [inline manifests/storage.pp:20]: I could not find any reference to $bacula::storage::storage.  Am missing something or is this parameter just unused?

**Comment by smortex** [comment]
Just added REFERENCE.md to the repo… Live preview here:
https://github.com/xaque208/puppet-bacula/blob/doc/REFERENCE.md

**Comment by xaque208** [comment]
This is looking great, thank you @smortex.

**Comment by smortex** [comment]
Cool ! This seems fair so far :-) @xaque208 i have not dig in the Travis-CI failure, the failing test having some "DEPLOY_TO_FORGE=yes" makes me feel good to see it failing…  Is this something that is completely fine and safe and that we can fix, or is something missing, e.g. something that ensure this is running only when a new tag is pushed?

**Comment by xaque208** [comment]
Oh I always forget to disable that bit when I rebase the modulesync repo, but yes disabling that entirely has been my aim.  Though, I could probably be convinced otherwise considering how bad I've been about cutting releases.

**Comment by smortex** [comment]
I never released anything to the forge (yet… it's still something we intend to do for some or the opus-codium modules), so I can't really give advice here unfortunately, but tell me if there is anything I can do that would help.

**Comment by xaque208** [comment]
I synced some testing changes to master that might help the tests along here.

**Comment by smortex** [comment]
Nice! I rebased my branch on top of master and it seems to be good :-)

File: manifests/storage.pp (hunk 1 above)
Comment (xaque208) [reply manifests/storage.pp:22]: Hmm, I don't see it either.  Must be left over from something.

Comment (smortex) [reply manifests/storage.pp]: Okay, it's used :rofl: 

```
Error: Could not retrieve catalog from remote server: Error 500 on SERVER: Server Error: Evaluation Error: Unknown variable: 'storage'. (file: /usr/local/etc/puppet/environments/production/modules/bacula/manifests/storage.pp, line: 95, column: 33)
```

Let's document this properly!

Comment (smortex) [reply manifests/storage.pp]: _The address to be configured on the director to communicate with this storage server_

**Comment by smortex** [comment]
So, the "unused" parameter was used :-)  I rebased the commits on top of master and ameded them to fix this "unused" parameter.

**Comment by xaque208** [comment]
Nice.  This is great, thank you for your efforts here.

**Pull request merged**